react-app/src/components/projects/AddProject.tsx

```diff
@@ -0,0 +1,375 @@
+import {
+  Autocomplete,
+  Box,
+  Button,
+  Grid,
+  IconButton,
+  InputAdornment,
+  TextField,
+  Typography,
+  useTheme,
+} from '@mui/material';
+import React, { useState } from 'react';
+import { NavigateBackButton } from '../display/DetailViewNavigation';
+import TextFormField from '../form/TextFormField';
+import AutocompleteFormField from '../form/AutocompleteFormField';
+import { FormProvider, useForm } from 'react-hook-form';
+import SingleSelectBoxFormField from '../form/SingleSelectBoxFormField';
+import { DataGrid, GridColDef } from '@mui/x-data-grid';
+import { Delete, Search } from '@mui/icons-material';
+import { Link } from 'react-router-dom';
+import usePimsApi from '@/hooks/usePimsApi';
+import { Parcel, ParcelEvaluation } from '@/hooks/api/useParcelsApi';
+import { Building, BuildingEvaluation } from '@/hooks/api/useBuildingsApi';
+import useDataLoader from '@/hooks/useDataLoader';
+
+interface IDisposalProjectSearch {
+  rows: any[];
+  setRows: (a: any[]) => void;
+}
+
+type ParcelWithType = Parcel & {
+  Type: string;
+};
+
+type BuildingWithType = Building & {
+  Type: string;
+};
+
+const DisposalProjectSearch = (props: IDisposalProjectSearch) => {
+  const { rows, setRows } = props;
+  const [autoCompleteVal, setAutoCompleteVal] = useState(null);
+  const [fuzzySearchOptions, setFuzzySearchOptions] = useState([]);
+  const [loadingOptions, setLoadingOptions] = useState(false);
+  const api = usePimsApi();
+  const theme = useTheme();
+
+  const getAutoCompleteLabel = (input: ParcelWithType | BuildingWithType | string) => {
+    if (typeof input === 'string') {
+      return '';
+    }
+
+    if (input.PID) {
+      return `${input.Type} - PID: ${input.PID}`;
+    } else if (input.PIN) {
+      return `${input.Type} - PIN: ${input.PIN}`;
+    } else if (input.Address1) {
+      return `${input.Type} - Address: ${input.Address1}`;
+    } else {
+      return `${input.Type} - No identifying info.`;
+    }
```

Comment:
If we get a result with no PID/PIN/Address, should we really show an option?

Reply:
This is just here to complete the logic block, realistically this should never actually happen.